src/main/java/bio/terra/service/dataset/flight/ingest/DatasetIngestFlight.java

```diff
@@ -27,19 +55,119 @@ public DatasetIngestFlight(FlightMap inputParameters, Object applicationContext)
     DatasetDao datasetDao = appContext.getBean(DatasetDao.class);
     DatasetService datasetService = appContext.getBean(DatasetService.class);
     BigQueryPdao bigQueryPdao = appContext.getBean(BigQueryPdao.class);
+    GcsPdao gcsPdao = appContext.getBean(GcsPdao.class);
     FireStoreDao fileDao = appContext.getBean(FireStoreDao.class);
     ConfigurationService configService = appContext.getBean(ConfigurationService.class);
     ApplicationConfiguration appConfig = appContext.getBean(ApplicationConfiguration.class);
+    ProfileService profileService = appContext.getBean(ProfileService.class);
+    LoadService loadService = appContext.getBean(LoadService.class);
+    ResourceService resourceService = appContext.getBean(ResourceService.class);
+    DatasetBucketDao datasetBucketDao = appContext.getBean(DatasetBucketDao.class);
+    ConfigurationService configurationService = appContext.getBean(ConfigurationService.class);
+    JobService jobService = appContext.getBean(JobService.class);
+    GoogleProjectService projectService = appContext.getBean(GoogleProjectService.class);
 
     // get data from inputs that steps need
     UUID datasetId =
         UUID.fromString(inputParameters.get(JobMapKeys.DATASET_ID.getKeyName(), String.class));
+    Dataset dataset = datasetService.retrieve(datasetId);
 
     RetryRule lockDatasetRetry =
         getDefaultRandomBackoffRetryRule(appConfig.getMaxStairwayThreads());
 
+    AuthenticatedUserRequest userReq =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+
+    IngestRequestModel ingestRequest =
+        inputParameters.get(JobMapKeys.REQUEST.getKeyName(), IngestRequestModel.class);
+
+    RetryRule randomBackoffRetry =
+        getDefaultRandomBackoffRetryRule(appConfig.getMaxStairwayThreads());
+    RetryRule driverRetry = new RetryRuleExponentialBackoff(5, 20, 600);
+
+    var platform = CloudPlatformWrapper.of(dataset.getDatasetSummary().getStorageCloudPlatform());
+
+    var profileId =
+        Optional.ofNullable(ingestRequest.getProfileId()).orElse(dataset.getDefaultProfileId());
+
+    int driverWaitSeconds = configService.getParameterValue(ConfigEnum.LOAD_DRIVER_WAIT_SECONDS);
+    int loadHistoryWaitSeconds =
+        configService.getParameterValue(ConfigEnum.LOAD_HISTORY_WAIT_SECONDS);
+    int fileChunkSize = configService.getParameterValue(ConfigEnum.LOAD_HISTORY_COPY_CHUNK_SIZE);
+
+    String loadTag = inputParameters.get(LoadMapKeys.LOAD_TAG, String.class);
+
     addStep(new LockDatasetStep(datasetDao, datasetId, true), lockDatasetRetry);
     addStep(new IngestSetupStep(datasetService, configService));
+
+    if (ingestRequest.getFormat() == IngestRequestModel.FormatEnum.JSON) {
+
+      // Begin file + metadata load
```

Comment:
This is a lot of code to add inline, so much that you needed to add start and end comments. Instead of adding comments, moving the code to a method would be another way to isolate it and provide documentation, using the method name or javadoc.